octree_web_viewer/src/bin/points_web_viewer.rs

```diff
@@ -12,44 +12,99 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use octree_web_viewer::utils;
-use point_viewer::octree;
+use octree_web_viewer::backend_error::PointsViewerError;
+use octree_web_viewer::state::AppState;
+use octree_web_viewer::utils::start_octree_server;
 use std::path::PathBuf;
 use std::sync::Arc;
 use structopt::StructOpt;
 
 /// HTTP web viewer for 3d points stored in OnDiskOctrees
 #[derive(StructOpt, Debug)]
 #[structopt(name = "points_web_viewer", about = "Visualizing points")]
-struct CommandlineArguments {
+pub struct CommandLineArguments {
     /// The octree directory to serve, including a trailing slash.
-    octree_path: String,
+    /// this overrides the path_* options
+    #[structopt(name = "DIR")]
+    octree_path: Option<String>,
     /// Port to listen on.
     #[structopt(default_value = "5433", long = "--port")]
     port: u16,
     /// IP string.
     #[structopt(default_value = "127.0.0.1", long = "--ip")]
     ip: String,
+    /// instead of DIR: specify path prefix for octree dir
+    #[structopt(long = "--prefix")]
```

Comment:
Nit: You can skip the `--` everywhere in `long`

Reply:
Done